### PR TITLE
2.18 release

### PR DESCRIPTION
**Pull request by edmundnoble**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,33 @@
 # `chainweb-node` Changelog
 
+## 2.18 (2023-03-01)
+
+This version replaces all previous versions. Any prior version will stop working
+on **2023-02-23T00:00:00Z**. Node administrators must upgrade to this version
```
Comment (larskuhtz) [inline CHANGELOG.md:6]:
```suggestion
on **2023-03-02T00:00:00Z**. Node administrators must upgrade to this version
```

**Review by emilypi — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
+
+*   New /local endpoint preflight simulation API. (#1585, #1600)
+*   Support for Pact 4.6: (#1602)
+    *    New ZK native function support.
+    *    Better gas estimation.
+*   Warning deprecation system support in /local:
+    *    Allows the node to provide warnings for upcoming feature deprecations in Pact.
+    *    The same warnings are generated by the repl in Pact 4.6.
+*   Internal changes to support future chain database schema changes.
+*   Remove libtbb as a dependency.
```
Comment (larskuhtz) [inline CHANGELOG.md:24]: Do we have to update the README for this change?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,33 @@
...
+
+Bug fixes:
+
+*   Filter Module Cache for just `coin` contract. (#1548)
+*   Prevent table name clashes in module. (#1556)
+
```
Comment (larskuhtz) [inline CHANGELOG.md:30]:
```suggestion
When upgrading directly from chainweb-node version 2.17, please, also take a look at the changes in versions 2.17.1 and 2.17.2 below.

```

File: node/ChainwebNode.hs [hunk 4]
```diff
@@ -511,7 +511,7 @@ pkgInfoScopes =
 -- SERVICE DATE for version 2.17
```
Comment (larskuhtz) [inline node/ChainwebNode.hs:511]:
```suggestion
-- SERVICE DATE for version 2.18
```

File: src/Chainweb/Version.hs [hunk 5]
```diff
@@ -1027,8 +1027,8 @@ chainweb217Pact aoa v h = case aoa of
 -- | Pact changes for Chainweb 2.13
 --
 chainweb218Pact :: ChainwebVersion -> BlockHeight -> Bool
-chainweb218Pact Mainnet01 = (>= 3518107)
-chainweb218Pact Testnet04 = (>= 3036950)
+chainweb218Pact Mainnet01 = (>= 3_512_363) -- 2023-03-03 00:00:00+00:00
+chainweb218Pact Testnet04 = (>= 3_038_343) -- 2023-03-02 12:00:00+00:00
```
Comment (larskuhtz) [inline src/Chainweb/Version.hs:1031]: 👍

**Review by larskuhtz — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (edmundnoble) [reply CHANGELOG.md:24]: Already done

**Review by larskuhtz — APPROVED** [review]

**Comment by larskuhtz** [comment]
dependency updates for 2.18:

```diff
diff --git a/cabal.project.freeze b/cabal.project.freeze
index 911868e27..3ab313d5b 100644
--- a/cabal.project.freeze
+++ b/cabal.project.freeze
@@ -57,7 +57,7 @@ constraints: any.Boolean ==0.2.4,
              any.blaze-markup ==0.8.2.8,
              any.boring ==0.2,
              boring +tagged,
-             any.bound ==2.0.5,
+             any.bound ==2.0.6,^M
              bound +template-haskell,
              any.bsb-http-chunked ==0.0.0.4,
              any.byteorder ==1.0.4,
@@ -92,7 +92,7 @@ constraints: any.Boolean ==0.2.4,
              any.containers ==0.6.5.1,
              any.contravariant ==1.5.5,
              contravariant +semigroups +statevar +tagged,
-             any.cookie ==0.4.5,
+             any.cookie ==0.4.6,^M
              any.criterion ==1.5.13.0,
              criterion -embed-data-files -fast,
              any.criterion-measurement ==0.1.4.0,
@@ -115,7 +115,7 @@ constraints: any.Boolean ==0.2.4,
              any.dense-linear-algebra ==0.1.0.0,
              any.deriving-compat ==0.6.2,
              deriving-compat +base-4-9 +new-functor-classes +template-haskell-2-11,
-             any.digraph ==0.2.2,
+             any.digraph ==0.3.0,^M
              any.direct-sqlite ==2.3.27,
              direct-sqlite +fulltextsearch +haveusleep +json1 -systemlib +urifilenames,
              any.directory ==1.3.6.0,
@@ -128,24 +128,26 @@ constraints: any.Boolean ==0.2.4,
              any.errors ==2.3.0,
              any.ethereum ==0.1.0.0,
              any.exceptions ==0.10.4,
-             any.fast-logger ==3.1.1,
+             any.fast-logger ==3.2.0,^M
              any.file-embed ==0.0.15.0,
              any.filepath ==1.4.2.1,
              any.fingertree ==0.1.5.0,
+             any.finite-typelits ==0.1.6.0,^M
              any.free ==5.1.10,
-             any.generic-data ==1.0.0.0,
+             any.generic-data ==1.0.0.1,^M
              generic-data -enable-inspect,
              any.ghc-boot-th ==8.10.7,
              any.ghc-compact ==0.1.0.0,
              any.ghc-heap ==8.10.7,
              any.ghc-prim ==0.6.1,
+             any.groups ==0.5.3,^M
              any.hashable ==1.3.0.0,
              hashable -examples +integer-gmp +sse2 -sse41,
              any.hashes ==0.2.3,
              hashes -benchmark-cryptonite -openssl-use-pkg-config -test-cryptonite +with-openssl,
              any.heaps ==0.4,
              any.hourglass ==0.2.12,
-             any.hsc2hs ==0.68.8,
+             any.hsc2hs ==0.68.9,^M
              hsc2hs -in-ghc-tree,
              any.http-api-data ==0.5,
              http-api-data -use-text-show,
@@ -155,8 +157,9 @@ constraints: any.Boolean ==0.2.4,
              any.http-date ==0.0.11,
              any.http-media ==0.8.0.0,
              any.http-types ==0.12.3,
-             any.http2 ==3.0.3,
+             any.http2 ==4.0.0,^M
              http2 -devel -doc -h2spec,
+             any.indexed-list-literals ==0.2.1.3,^M
              any.indexed-traversable ==0.1.2,
              any.indexed-traversable-instances ==0.1.1.1,
              any.integer-gmp ==1.0.3.0,
@@ -174,8 +177,8 @@ constraints: any.Boolean ==0.2.4,
              libyaml -no-unicode -system-libyaml,
              any.lifted-base ==0.2.3.12,
              any.loglevel ==0.1.0.0,
-             any.managed ==1.0.9,
-             any.massiv ==1.0.2.0,
+             any.managed ==1.0.10,^M
+             any.massiv ==1.0.3.0,^M
              massiv -unsafe-checks,
              any.math-functions ==0.3.4.2,
              math-functions +system-erf +system-expm1,
@@ -185,9 +188,11 @@ constraints: any.Boolean ==0.2.4,
              memory +support_bytestring +support_deepseq,
              any.merkle-log ==0.1.1,
              any.microlens ==0.4.13.1,
-             any.microstache ==1.0.2.2,
+             any.microstache ==1.0.2.3,^M
              any.mime-types ==0.1.1.0,
              any.mmorph ==1.2.0,
+             any.mod ==0.1.2.2,^M
+             mod +semirings +vector,^M
              any.monad-control ==1.0.3.1,
              any.mono-traversable ==1.0.15.3,
              any.mtl ==2.2.2,
@@ -198,7 +203,7 @@ constraints: any.Boolean ==0.2.4,
              any.network ==3.1.2.7,
              network -devel,
              any.network-byte-order ==0.1.6,
-             any.network-uri ==2.6.4.1,
+             any.network-uri ==2.6.4.2,^M
              any.newtype-generics ==0.6.2,
              any.nothunks ==0.1.3,
              nothunks +bytestring +text +vector,
@@ -217,6 +222,8 @@ constraints: any.Boolean ==0.2.4,
              any.parsers ==0.12.11,
              parsers +attoparsec +binary +parsec,
              any.pem ==0.2.4,
+             any.poly ==0.5.1.0,^M
+             poly +sparse,^M
              any.pretty ==1.1.3.6,
              any.prettyprinter ==1.6.0,
              prettyprinter -buildreadme,
@@ -229,7 +236,7 @@ constraints: any.Boolean ==0.2.4,
              any.quickcheck-instances ==0.3.28,
              quickcheck-instances -bytestring-builder,
              any.random ==1.2.1.1,
-             any.recv ==0.0.0,
+             any.recv ==0.1.0,^M
              any.reducers ==3.12.4,
              any.reflection ==2.1.6,
              reflection -slow +template-haskell,
@@ -238,6 +245,7 @@ constraints: any.Boolean ==0.2.4,
              any.retry ==0.9.3.0,
              retry -lib-werror,
              any.rocksdb-haskell-kadena ==1.1.0,
+             rocksdb-haskell-kadena -with-tbb,^M
              any.rosetta ==1.0.1,
              any.rts ==1.0.1,
              any.safe ==0.3.19,
@@ -250,6 +258,8 @@ constraints: any.Boolean ==0.2.4,
              semigroupoids +comonad +containers +contravariant +distributive +tagged +unordered-containers,
              any.semigroups ==0.20,
              semigroups +binary +bytestring -bytestring-builder +containers +deepseq +hashable +tagged +template-haskell +text +transformers +unordered-containers,
+             any.semirings ==0.6,^M
+             semirings +containers +unordered-containers,^M
              any.servant ==0.19.1,
              any.servant-client ==0.19,
              any.servant-client-core ==0.19,
@@ -319,7 +329,7 @@ constraints: any.Boolean ==0.2.4,
              unix-compat -old-time,
              any.unix-time ==0.4.8,
              any.unliftio ==0.2.23.0,
-             any.unliftio-core ==0.2.0.1,
+             any.unliftio-core ==0.2.1.0,^M
              any.unordered-containers ==0.2.15.0,
              unordered-containers -debug,
              any.utf8-string ==1.0.2,
@@ -331,7 +341,9 @@ constraints: any.Boolean ==0.2.4,
              any.vector-algorithms ==0.9.0.1,
              vector-algorithms +bench +boundschecks -internalchecks -llvm +properties -unsafechecks,
              any.vector-binary-instances ==0.2.5.2,
+             any.vector-sized ==1.5.0,^M
              any.vector-space ==0.16,
+             any.vector-stream ==0.1.0.0,^M
              any.vector-th-unbox ==0.2.2,
              any.void ==0.7.3,
              void -safe,
@@ -343,18 +355,18 @@ constraints: any.Boolean ==0.2.4,
              wai-extra -build-example,
              any.wai-logger ==2.4.0,
              any.wai-middleware-throttle ==0.3.0.1,
-             any.warp ==3.3.23,
+             any.warp ==3.3.24,^M
              warp +allow-sendfilefd -network-bytestring -warp-debug +x509,
-             any.warp-tls ==3.3.4,
+             any.warp-tls ==3.3.5,^M
              any.word8 ==0.1.3,
              any.x509 ==1.7.7,
              any.x509-store ==1.6.9,
              any.x509-system ==1.6.7,
              any.x509-validation ==1.6.12,
-             any.yaml ==0.11.8.0,
+             any.yaml ==0.11.10.0,^M
              yaml +no-examples +no-exe,
              any.yet-another-logger ==0.4.1,
              yet-another-logger -tbmqueue,
              any.zlib ==0.6.3.0,
              zlib -bundled-c-zlib -non-blocking-ffi -pkg-config
-index-state: hackage.haskell.org 2023-01-24T15:49:59Z
+index-state: hackage.haskell.org 2023-02-14T18:10:13Z^M
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,34 @@
...
+*   Warning deprecation system support in /local:
+    *    Allows the node to provide warnings for upcoming feature deprecations in Pact.
+    *    The same warnings are generated by the repl in Pact 4.6.
+*   Internal changes to support future chain database schema changes.
+*   Remove libtbb as a dependency.
+
+Bug fixes:
+
+*   Filter Module Cache for just `coin` contract. (#1548)
+*   Prevent table name clashes in module. (#1556)
```
Comment (jwiegley) [inline CHANGELOG.md:29]: @larskuhtz Can you please add to this list:
```suggestion
*   Prevent table name clashes in module. (#1556)
*   Full chain replay is now possible on Intel Mac & Linux, and M1 Mac.
```

**Review by jwiegley — APPROVED** [review]

**Pull request merged**